srml/support/src/traits.rs

```diff
@@ -91,16 +91,20 @@ pub enum UpdateBalanceOutcome {
 }
 
 /// Simple trait designed for hooking into a transaction payment.
-///
-/// It operates over a single generic `AccountId` type.
-pub trait MakePayment<AccountId> {
+pub trait MakePayment<AccountId, Balance> {
 	/// Make transaction payment from `who` for an extrinsic of encoded length
 	/// `encoded_len` bytes. Return `Ok` iff the payment was successful.
 	fn make_payment(who: &AccountId, encoded_len: usize) -> Result<(), &'static str>;
+
+	/// Make a raw transactiona payment from `who` with the value `value`.
+	/// This is most often used to deduct optional fees from a transactor.
+	/// Return `Ok` iff the payment was successful.
+	fn make_raw_payment(who: &AccountId, value: Balance) -> Result<(), &'static str>;
```

Comment:
maybe let's just call it `make_tip_payment` and avoid being too generic?

Reply:
Will probably refactor this but in terms of being generic it won't help since the tip itself is also generic over `balance`.

Reply:
I meant just the name - `make_raw_payment` is more generic than `make_tip_payment`, which immediately tells you about the context it's used in.